sdk/keyvault/azure-keyvault-keys/azure/keyvault/keys/crypto/_client.py

```diff
@@ -180,11 +191,13 @@ def encrypt(self, algorithm, plaintext, **kwargs):
         _validate_arguments(operation=KeyOperation.encrypt, algorithm=algorithm, iv=iv, aad=aad)
         self._initialize(**kwargs)
 
-        if self._local_provider.supports(KeyOperation.encrypt, algorithm):
+        if self._local_provider.supports(KeyOperation.encrypt, algorithm) or self._local_only:
```

Comment:
That's a good point -- I was on the fence about this, since it is counterintuitive (and potentially bad, as you point out) to proceed with an operation that we know isn't supported. The intention was to raise the resulting error message -- and the minimal refactoring was appealing -- but I agree that it's probably better to raise the NotImplementedError here